packages/labs/ssr-react/src/lib/node/wrap-create-element.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {isCustomElement} from '../utils.js';
+import {renderShadowContents} from './render-shadow-contents.js';
+
+import type {
+  createElement as ReactCreateElement,
+  ElementType,
+  ReactElement,
+  ReactNode,
+} from 'react';
+
+/**
+ * Wraps the provided `createElement` function to also server render Lit
+ * component's shadow DOM.
+ */
+export function wrapCreateElement(
+  // TODO(augustjk) Should the type for this param be more generic to allow
+  // non-React alternatives like preact?
+  originalCreateElement: typeof ReactCreateElement
+) {
+  return function createElement<P extends {}>(
+    type: ElementType<P>,
+    props: P,
+    ...children: ReactNode[]
+  ): ReactElement {
+    if (isCustomElement(type)) {
+      const {shadowContents, elementAttributes, templateAttributes} =
+        renderShadowContents(type, props);
+
+      if (shadowContents !== undefined) {
+        const templateShadowRoot = originalCreateElement('template', {
+          ...templateAttributes,
+          dangerouslySetInnerHTML: {
+            __html: [...shadowContents].join(''),
+          },
+        });
+
+        return originalCreateElement(
```

Comment:
Should be outside the `if (shadowContents !== undefined)` test? We should still render a CE that doesn't render a SR for whatever reason.

Reply:
It would fall back to the `return originalCreateElement(type, props, ...children)` outside the `if (isCustomElement(type))` block in that case.